remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/LifecycleManager.java

```diff
@@ -247,13 +244,14 @@ private QuerySerializers buildQuerySerializers(ComponentRegistry cr, MediaType s
 
    @Override
    public void cacheStarted(ComponentRegistry cr, String cacheName) {
-      InternalCacheRegistry icr = cr.getGlobalComponentRegistry().getComponent(InternalCacheRegistry.class);
+      GlobalComponentRegistry gcr = cr.getGlobalComponentRegistry();
+      InternalCacheRegistry icr = gcr.getComponent(InternalCacheRegistry.class);
       if (!icr.isInternalCache(cacheName)) {
          Configuration cfg = cr.getComponent(Configuration.class);
-         ProtobufMetadataManagerImpl protobufMetadataManager = (ProtobufMetadataManagerImpl) cr.getGlobalComponentRegistry().getComponent(ProtobufMetadataManager.class);
-         SerializationContext serCtx = protobufMetadataManager.getSerializationContext();
-
          if (cfg.indexing().index().isEnabled()) {
+            ProtobufMetadataManagerImpl protobufMetadataManager = (ProtobufMetadataManagerImpl) gcr.getComponent(ProtobufMetadataManager.class);
+            SerializationContext serCtx = protobufMetadataManager.getSerializationContext();
```

Comment:
`cacheStarted` has a similar problem with `cacheManagerStarted`: if 2 threads call `getCache(X)`, one of them will actually start the cache and invoke `cacheStarted` in all the modules. The other thread will resume as soon as the cache has been started, without waiting for the modules' `cacheStarted` code to run.